plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -340,6 +313,62 @@ public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHand
         }
     }
 
+    private Optional<JdbcColumnHandle> getColumn(ConnectorSession session, SchemaTableName schemaTableName, Connection connection, ResultSet resultSet)
+            throws SQLException
+    {
+        String columnName = resultSet.getString("COLUMN_NAME");
+        try {
+            return getColumn(session, schemaTableName, connection, resultSet, columnName);
+        }
+        catch (SQLException | RuntimeException e) {
+            throw new TrinoException(JDBC_ERROR, "Failed to resolve column %s in table %s. %s".formatted(columnName, schemaTableName, firstNonNull(e.getMessage(), e)), e);
```

Comment:
Am I right, that this catch is the main change (with proper error message)? and other is more refactoring?

Reply:
Yes. Other is just a way to encapsulate logic to retrieve a column, which could be wrapped in not that ugly way in try-catch.